rpc/wrtc_signaling_answerer.go

```diff
@@ -29,15 +29,21 @@ const testDelayAnswererNegotiationVar = "TEST_DELAY_ANSWERER_NEGOTIATION"
 type webrtcSignalingAnswerer struct {
 	startStopMu sync.Mutex // startStopMu guards the Start and Stop methods so they do not happen concurrently.
 
-	address                 string
-	hosts                   []string
-	server                  *webrtcServer
-	dialOpts                []DialOption
-	webrtcConfig            webrtc.Configuration
-	activeBackgroundWorkers sync.WaitGroup
-	cancelBackgroundWorkers func()
-	closeCtx                context.Context
-	logger                  golog.Logger
+	address      string
+	hosts        []string
+	server       *webrtcServer
+	dialOpts     []DialOption
+	webrtcConfig webrtc.Configuration
+
+	// answererMu should be `Lock`ed in `Stop` to `Wait` on ongoing answer workers in startAnswerer/answer.
+	// answererMu should be `RLock`ed when starting a new answer worker (allow concurrent answer workers) to
+	// `Add` to answerWorkers.
+	answerMu            sync.RWMutex
```

Comment:
q: can this mutex replace `startStopMu`? or is it necessary to have separate mutexes for starting/stopping vs in-progress answers.

Reply:
> `Start` should never interfere with in progress answers

q:  just to clarify - do mean _should_ as in start and in-progress will never interfere w/ each other, or they might and it's a requirement that they don't? if the former than i think one mutex makes things simpler/clearer, if the latter then separate mutexes makes sense.

Reply:
Update: `Start` and `startAnswerer` do indeed interfere with each other. If we were to use one mutex, `Start` would `Lock` it, call `startAnswerer`, and `startAnswerer` would `Rock()` the same lock to `Add` to the `WaitGroup` and deadlock the system 😢 . So, I think we need two muteness.

Reply:
gotcha, thanks for looking into this. makes sense!